### ISSUE TITLE
Add Credentials setting

### ISSUE THREAD

**Issue by rsuter** [NSwag/NSwag#734]
Add "Credentials" setting to add the "credentials" options in the window.fetch template.

Type: string

**Comment by grovesNL** [comment]
I transform the headers similarly to add an OAuth token for example. Not sure if headers should be added first class too if credentials are supported.

**Comment by rsuter** [comment]
Yes, I currently also transform it - this should be used for more "complex" scenarios... this one is just for this simple case - but i'm not yet sure if this is a good idea...

**Comment by ZhaoRd** [comment]
How can use "Credentials" in  .nswag ?

**Comment by jeremyVignelles** [comment]
Hi,
What's the status of this?

It's quite cumbersome to have that code for each nswag generated client:
```ts
export class UseHttpCookiesForApi {
    protected transformOptions(options: RequestInit): Promise<RequestInit> {
        options.credentials = 'same-origin';
        return Promise.resolve(options);
    }
}
```
